### ISSUE TITLE
Remove deprecated lifecycle method componentWillReceiveProps

### ISSUE THREAD

**Issue by verhey** [howlowck/react-powerbi#12]
**Using the component with a version of React greater than 16.9 will produce the following warning:** 

```
 Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-async-component-lifecycle-hooks for details.
```

https://github.com/howlowck/react-powerbi/blob/22c21d4c3601e2873a78b5b2f145a22d645952fb/src/index.js#L23

This is due to changes introduced in React 16.3 and can be addressed by appending an `UNSAFE_` to the lifecycle method, or refactoring the component to use a different lifecycle method. 

React's suggestions on mitigating the changes can be found [here](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html).

**Suggestions:** 
* Append the `UNSAFE_` string to the method. 
  * This will at least allow users to update to react 17.0 when it releases
  * Can be done via `npx react-codemod rename-unsafe-lifecycles`
* Refactor component state management to remove dependency on `componentWillReceiveProps `
  * At first glance this doesn't appear too bad, but I'm not particularly familiar with how the component state is managed, and my own implementation of the component is very simple. Without any test coverage I would be hesitant to introduce any potentially breaking changes.

**Comment by howlowck** [comment]
Thanks.. for now, i'm leaning towards simply appending the `UNSAFE_` string to the method. 
Do you want to make a PR for that?

**Comment by verhey** [comment]
Opened a PR. Note that because the "UNSAFE_" addition breaks eslint's `camelcase` rule present in `.eslintrc`. I chose to update eslint and use the `allow` option on the existing `camelcase` rule, but that can be reverted if you'd rather keep the PR as small as possible.

**Comment by howlowck** [comment]
I merged it and published it to npm react-powerbi@0.5.2
Thanks for your contribution! Keep it up 😄
